### PR TITLE
update the CHANGELOG with some missing changes.

### PR DESCRIPTION
**Pull request by krasi-georgiev**
missed to add these with some merged PRs

Signed-off-by: Krasi Georgiev <kgeorgie@redhat.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+   - Disabled by default and can be enabled via `AllowOverlappingBlock` option.
    - Added `MergeChunks` function in `chunkenc/xor.go` to merge 2 time-overlapping chunks.
    - Added `MergeOverlappingChunks` function in `chunks/chunks.go` to merge multiple time-overlapping Chunk Metas.
    - Added `MinTime` and `MaxTime` method for `BlockReader`.
- - [CHANGE]  `NewLeveledCompactor` takes a context so that a compaction is canceled when closing the db.
  - [ENHANCEMENT] When closing the db any running compaction will be cancelled so it doesn't block.
+   - `NewLeveledCompactor` takes a context.
  - [CHANGE] `prometheus_tsdb_storage_blocks_bytes_total` is now `prometheus_tsdb_storage_blocks_bytes`
+ - [FEATURE] New `dump` command to tsdb tool to dump all samples.
+ - [FEATURE] New `encoding` package for common binary encoding/decoding helpers.
```
Comment (brian-brazil) [inline CHANGELOG.md]: This sounds more like an ENHANCEMENT

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
 ## master / unreleased
  - [ENHANCEMENT] Time-ovelapping blocks are now allowed. [#370](https://github.com/prometheus/tsdb/pull/370)
```
Comment (brian-brazil) [inline CHANGELOG.md]: This is a FEATURE

File: CHANGELOG.md (hunk 1 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: all of these are exported so I would say any library user can use these. That is why I thought of it as a new feature.

File: CHANGELOG.md (hunk 2 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: updated, thanks!

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**